platform/darwin/src/MGLStyleLayerManager.mm

```diff
@@ -38,22 +38,32 @@
 }
 
 void LayerManagerDarwin::addLayerType(std::unique_ptr<LayerPeerFactory> factory) {
-    auto* coreFactory = factory->getCoreLayerFactory();
-    std::string type{coreFactory->getTypeInfo()->type};
+    assert(getFactory(factory->getCoreLayerFactory()->getTypeInfo()) == nullptr);
```

Comment:
In Obj-C/Obj-C++ code, generally prefer `NSAssert` and `NSCAssert` over `assert`. 

However, we recently added an `MGLAssert` and `MGLCAssert` that also optionally logs - it's in `MGLLoggingConfiguration_Private.h` - can you use that here (and for the other asserts)?

Reply:
I'm not sure log is needed here, this assertion is rather to ensure the `LayerManager` API contract.

Reply:
In that case, let’s use `NSCAssert()`.